src/components/TokenDisplay.tsx

```diff
@@ -54,9 +54,6 @@ function TokenDisplay({
   style,
   testID,
 }: Props) {
-  if (tokenAddress ? currency : !currency) {
-    throw new Error('TokenDisplay must be passed either "currency" or "tokenAddress" and not both')
```

Comment:
I'm lacking some context on why this was here to begin with. but if tokenAddress is now optional, you could still throw if both are provided

Reply:
though just using discriminated union type for the params would be nicer, since it would result in a compile-time check for this rather than a runtime error

Reply:
Using a discriminated union would be nicer, but it would mean we would need two new types e.g.

```ts
interface Address {
  kind: 'address',
  tokenAddress: string
}

interface Currency {
  kind: 'currency',
  currency: string
} 

type AddressOrCurrency = Address | Currency
```
which would require changes to a ton of files that use this component, so I think this is out of scope for this PR. This now throws an error when both are provided, however.

Reply:
I think you could do it with just this
```
interface Address {
  tokenAddress: string
}
interface Currency {
  currency: string
}
type AddressOrCurrency = Address | Currency
```
but yeah not that important